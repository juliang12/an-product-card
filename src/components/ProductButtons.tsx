import React, { useContext, useCallback } from 'react';
import styles from '../styles/styles.module.css';
import { ProductContext } from './ProductCard';

export interface ButtonsProps {
  className?: string;
  style?: React.CSSProperties;
}

export const ProductButtons = ({ className, style }: ButtonsProps) => {
  const { increaseBy, counter, maxCount } = useContext(ProductContext);

  const isMaxReach = useCallback(() => {
    if (counter === maxCount) {
      return true;
    } else {
      return false;
    }
  }, [counter, maxCount]);

  return (
    <div className={`${styles.buttonsContainer} ${className}`} style={style}>
      <button onClick={() => increaseBy(-1)} className={styles.buttonMinus}>
        -
      </button>
      <div className={styles.countLabel}>{counter}</div>
      <button
        disabled={isMaxReach()}
        onClick={() => increaseBy(1)}
        className={`${styles.buttonAdd} ${isMaxReach() && styles.disabled}`}
      >
        +
      </button>
    </div>
  );
};
