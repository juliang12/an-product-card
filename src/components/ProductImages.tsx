import React, { useContext } from 'react';
import styles from '../styles/styles.module.css';
import noImg from '../assets/no-image.jpg';
import { ProductContext } from './ProductCard';

export interface ProductImageProps {
  img?: string;
  className?: string;
}

export const ProductImage = ({ img, className }: ProductImageProps) => {
  const { product } = useContext(ProductContext);
  let imgToShow: string;

  if (img) {
    imgToShow = img;
  } else if (product.img) {
    imgToShow = product.img;
  } else {
    imgToShow = noImg;
  }
  return (
    <img
      className={`${styles.productImg} ${className}`}
      src={imgToShow}
      alt=""
    />
  );
};
