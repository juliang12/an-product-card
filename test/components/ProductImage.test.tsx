import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage } from '../../src/components';

const product1 = {
  id: '1',
  title: 'Coffee Mug',
  img: './coffee-mug.png',
};

describe('ProductImage', () => {
  test('debe de mostrar el componente correctamente', () => {
    const wrapper = renderer.create(
      <ProductImage img={product1.img} className="class custom" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
