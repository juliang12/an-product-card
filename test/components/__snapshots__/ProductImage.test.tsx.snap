// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente correctamente 1`] = `
<img
  alt=""
  className="productImg class custom"
  src="./coffee-mug.png"
/>
`;
